subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/PomReader.java

```diff
@@ -684,6 +697,17 @@ private boolean isActivationPropertyActivated(Element propertyElement) {
         return getPomProperties(projectElement);
     }
 
+    /**
+     * @return properties of both current and children poms.
+     */
+    public Map<String, String> getAllPomProperties() {
```

Comment:
(I copy/paste it from the outdated commit)
The maybeSetGavProperties and setParentPomProperties put in properties
something else than what is inside the <properties> tag of the pom.
The purpose of the set of property keys is to record what is in the <properties> tag, as these are the only ones that should be overridden in parent/grand-parent poms.